swifttool/manager.py

```diff
@@ -0,0 +1,282 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# Copyright (c) 2014, Blue Box Group, Inc.
+# Copyright (c) 2014, Craig Tracey <craigtracey@gmail.com>
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+# implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+#
+
+import copy
+import logging
+import math
+import os
+import shutil
+import time
+import yaml
+from swift.common.ring import RingBuilder
+
+from swifttool.ring_defintion import ringsdef_helper
+from swifttool.utils import RING_TYPES, get_devices
+
+LOG = logging.getLogger(__name__)
+
+_STATUS_FILE = 'capacity.status'
+
+
+class CapacityManager(object):
+
+    def __init__(self, iterations=10,
+                 def_file='/etc/swift/ring_definition.yml',
+                 scaleup=True, metadata=None,
+                 outdir='/etc/swift/'):
+        self.iterations = iterations
+        self.def_file = def_file
+        self.scaleup = scaleup
+        self.cluster_changes = {}
+        self.outdir = outdir
+        self.cluster_config = None
+        self.metadata = metadata
+        self.current_status = os.path.join(outdir, _STATUS_FILE)
+        self._calculate_cluster_changes()
+
+    def __del__(self):
+        # Cleanup the files we create
+        os.remove(self.current_status)
+        os.remove(self.def_file + ".orig")
+
+    def update_cluster(self):
+        LOG.info("Starting to update the cluster")
+        # Save the copy
+        shutil.copyfile(self.def_file, self.def_file + ".orig")
+        self._write_status(0)
+        min_part_hours = int(self._get_builder('account').min_part_hours)
+        for iteration in range(1, self.iterations + 1):
+            changed, cfg = self._update_cluster_nodes()
+            if not changed:
+                break
+            # Write up the new config in /etc/swift
+            with open(self.def_file, 'w') as stream:
+                yaml.dump(cfg, stream, default_flow_style=False,
+                          explicit_start=True)
+            LOG.info("Rings distributed. Sleeping for %d hours 1 minutes" % min_part_hours)
+            # Sleep for min_part_hours + minute
+            time.sleep((min_part_hours * 60 * 60) + 60)
+            # Check if rings need to be rebalanced
+            self._rebalance_rings(cfg, min_part_hours)
+            # Update status
+            self._write_status(int((100 * iteration) / self.iterations))
+            LOG.info("Iteration %d complete" % iteration)
```

Comment:
This is great! but just FYI, you could even treat it like a str and it'll still work, even in py3k: 
```
In [49]: "Watch me cast an int implicitly! %s" % 54321
Out[49]: 'Watch me cast an int implicitly! 54321'
```